Src/IronPython.Modules/ModuleOps.cs

```diff
@@ -125,25 +128,26 @@ public static partial class ModuleOps {
 
         public static CTypes.CData TryCheckCDataPointerType(object o, object type) {
             CTypes.CData res = o as CTypes.CData;
-            object asParam;
-            if (res == null && PythonOps.TryGetBoundAttr(o, "_as_parameter_", out asParam)) {
+            if (res == null && PythonOps.TryGetBoundAttr(o, "_as_parameter_", out object asParam)) {
                 res = asParam as CTypes.CData;
             }
-            if (res != null && res.NativeType != ((CTypes.PointerType)type)._type) {
-                if (!(res is CTypes.Pointer)) {
-                    throw ArgumentError(type, ((PythonType)((CTypes.PointerType)type)._type).Name, o);
-                }
-                return null;
+
+            bool valid = true;
+            if (res.NativeType is CTypes.ArrayType at) {
```

Comment:
If `res == null` will this fail? Also as above, not clear what check is going on.